Sources/WhoopDIKit/Container/Container.swift

```diff
@@ -54,28 +60,30 @@ public final class Container {
     public func inject<T>(_ name: String? = nil,
                           params: Any? = nil,
                           _ localDefinition: (DependencyModule) -> Void) -> T {
-        guard localServiceDict == nil else {
+        guard !isLocalInjectActive else {
             fatalError("Nesting WhoopDI.inject with local definitions is not currently supported")
         }
+        
+        isLocalInjectActive = true
+        defer {
+            isLocalInjectActive = false
+            localDependencyGraph.resetDependencyGraph()
+        }
         // We need to maintain a reference to the local service dictionary because transient dependencies may also
         // need to reference dependencies from it.
         // ----
-        // This is a little dangerous since we are mutating a static variable but it should be fine as long as you
+        // This is a little dangerous since we are mutating a variable but it should be fine as long as you
         // don't use `inject { }` within the scope of another `inject { }`.
-        let serviceDict = ServiceDictionary<DependencyDefinition>()
-        localServiceDict = serviceDict
-        defer {
-            localServiceDict = nil
-        }
-
-        let localModule = DependencyModule()
-        localDefinition(localModule)
-        localModule.addToServiceDictionary(serviceDict: serviceDict)
-
-        do {
-            return try get(name, params)
-        } catch {
-            fatalError("WhoopDI inject failed with error: \(error)")
+        return localDependencyGraph.aquireDependencyGraph { localServiceDict in
+            let localModule = DependencyModule()
+            localDefinition(localModule)
+            localModule.addToServiceDictionary(serviceDict: localServiceDict)
+            
+            do {
+                return try get(name, params)
+            } catch {
+                fatalError("WhoopDI inject failed with error: \(error)")
```

Comment:
While we are here, can we pack more info into this error message. i.e. the name and params. If we don't already print this somewhere